src/renderer/components/Gallery.tsx

```diff
@@ -117,15 +186,60 @@ export default class Gallery extends Component<
     }
   }
 
+  updateFirstVisibleMessage(message: Type.MessageLoadResult) {
+    if (message.variant === 'message') {
+      if (this.dateHeader.current)
+        this.dateHeader.current.innerText = moment(
+          message.timestamp * 1000
+        ).format('LL')
+    }
+  }
+
+  openFullscreenMedia(message: Type.Message) {
+    window.__openDialog(FullscreenMedia, {
+      msg: message,
+      neighboringMedia:
+        this.props.chatId === 'all'
+          ? NeighboringMediaMode.Global
+          : NeighboringMediaMode.Chat,
+    })
+  }
+
   render() {
-    const { medias, id, errors } = this.state
+    const {
+      mediaMessageIds,
+      mediaLoadResult,
+      id,
+      loading,
+      queryText,
+      GalleryImageKeepAspectRatio,
+    } = this.state
     const tx = window.static_translate // static because dynamic isn't too important here
     const emptyTabMessage = this.emptyTabMessage(id)
 
+    const filteredMediaMessageIds = mediaMessageIds.filter(id => {
+      const result = mediaLoadResult[id]
+      if (
+        result.variant === 'message' &&
+        result.fileName?.indexOf(queryText) !== -1
```

Comment:
We should open an issue to find a more efficient algorithm for the checking.

Reply:
And finding a good algorithm is not far away. I'm pretty sure an npm package already exists for it :)

Reply:
indexOf of is already one of the fastest methods... anyway this needs to be done in core, because currently this is one of the things that forces us to load all file informations from the DB (not just their message ids). If you have many images you might feel the delay when you go into the image tab, because currently it loads every message object from core, the chatlist on the other hand only loads visible chatlistentry objects from core. see also #3446